src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -5,14 +5,1181 @@
 #include "TerminalPage.h"
 
 #include "TerminalPage.g.cpp"
+#include <winrt/Microsoft.UI.Xaml.XamlTypeInfo.h>
 
 using namespace winrt;
 using namespace Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::ApplicationModel::DataTransfer;
+using namespace winrt::Windows::UI::Text;
+using namespace winrt::Microsoft::Terminal;
+using namespace winrt::Microsoft::Terminal::TerminalControl;
+using namespace winrt::Microsoft::Terminal::TerminalConnection;
+using namespace winrt::Microsoft::Terminal::Settings;
+using namespace ::TerminalApp;
+
+// Note: Generate GUID using TlgGuid.exe tool
+TRACELOGGING_DEFINE_PROVIDER(
+    g_hTerminalAppProvider,
```

Comment:
>g_hTerminalAppProvider [](start = 4, length = 22)

I'm not sure if this should be moved to TerminalPage.  It is the only user currently, so this may be fine.  But I worry that if a second TerminalPage gets created that it will stomp over the first.  The code doesn't do that currently, but the way this is currently defined opens up potential fragility.

Reply:
the terminal tl provider was moved into `DllMain` in `init.cpp` anyway :smile:

Reply:
Seems that TerminalPage can just consume it ? Let me do a sync today, thanks !